kms-connector/crates/utils/src/types/fhe.rs

```
@@ -7,58 +7,23 @@ use kms_grpc::kms::v1::TypedPlaintext;
-        t if t == FheTypes::Uint160 as i32 => "EUINT160",
-        t if t == FheTypes::Uint256 as i32 => "EUINT256",
-        t if t == FheTypes::Uint512 as i32 => "EUINT512",
-        t if t == FheTypes::Uint1024 as i32 => "EUINT1024",
-        t if t == FheTypes::Uint2048 as i32 => "EUINT2048",
-        _ => "UNKNOWN",
-    }
-}
-
 /// Extracts the FHE type from handle bytes.
-pub fn extract_fhe_type_from_handle(bytes: &[u8]) -> i32 {
+pub fn extract_fhe_type_from_handle(bytes: &[u8]) -> FheTypes {
     // Format: keccak256(keccak256(bundleCiphertext)+index)[0:29] + index + type + version
     // - Last byte (31): Version (currently 0)
     // - Second-to-last byte (30): FHE Type
     // - Third-to-last byte (29): Handle index
     // - Rest (0-28): Hash data
     if bytes.len() >= 32 {
         let type_byte = bytes[30]; // FHE type is at index 30
-
-        if type_byte >= 12 {
+        let Ok(fhe_type) = FheTypes::try_from(type_byte as i32) else {
             error!("Unknown FHE type byte: {}, must be less than 12", type_byte);
-            return FheTypes::Bool as i32;
-        }
-
-        match type_byte {
-            0 => FheTypes::Bool as i32,
-            1 => FheTypes::Uint4 as i32,
-            2 => FheTypes::Uint8 as i32,
-            3 => FheTypes::Uint16 as i32,
-            4 => FheTypes::Uint32 as i32,
-            5 => FheTypes::Uint64 as i32,
-            6 => FheTypes::Uint128 as i32,
-            7 => FheTypes::Uint160 as i32,
-            8 => FheTypes::Uint256 as i32,
-            9 => FheTypes::Uint512 as i32,
-            10 => FheTypes::Uint1024 as i32,
-            11 => FheTypes::Uint2048 as i32,
-            _ => unreachable!(), // We checked type_byte < 12 above
-        }
+            return FheTypes::Bool;
+        };
+        fhe_type
     } else {
         error!("Handle too short: {} bytes, expected 32 bytes", bytes.len());
-        FheTypes::Bool as i32
+        FheTypes::Bool
```

Maybe we could make this function return a Result instead due to the error that we swallow here?

---

Indeed! Should be fixed